pkgs/tools/inputmethods/ibus/default.nix

```diff
@@ -1,129 +1,141 @@
+      repo = "cldr-emoji-annotation";
+      rev = version;
+      sha256 = "1vsj32bg8ab4d80rz0fxy6sj2lv31inzyjnddjm079bnvlaf2kih";
     };
+    nativeBuildInputs = [ autoreconfHook ];
   };
-  pyEnv = python3.buildEnv.override {
-    extraLibs = [ python3.pkgs.pygobject3 ];
-
+  python3Runtime = python3.withPackages (pypkgs: with pypkgs; [ pygobject3 ]);
+  python3BuildEnv = python3.buildEnv.override {
     # ImportError: No module named site
     postBuild = ''
-      makeWrapper '${glib.dev}/bin/glib-genmarshal' "$out"/bin/glib-genmarshal \
-        --unset PYTHONPATH
+      makeWrapper ${glib.dev}/bin/gdbus-codegen $out/bin/gdbus-codegen --unset PYTHONPATH
+      makeWrapper ${glib.dev}/bin/glib-genmarshal $out/bin/glib-genmarshal --unset PYTHONPATH
+      makeWrapper ${glib.dev}/bin/glib-mkenums $out/bin/glib-mkenums --unset PYTHONPATH
     '';
   };
-in stdenv.mkDerivation rec {
+in
+
+stdenv.mkDerivation rec {
   name = "ibus-${version}";
-  version = "1.5.16";
+  version = "1.5.17";
 
-  src = fetchurl {
-    url = "https://github.com/ibus/ibus/releases/download/${version}/${name}.tar.gz";
-    sha256 = "07py16jb81kd7vkqhcia9cb2avsbg5jswp2kzf0k4bprwkxppd9n";
+  src = fetchFromGitHub {
+    owner = "ibus";
+    repo = "ibus";
+    rev = version;
+    sha256 = "09mrj9d8qpl9cbylg1zx8c3ymc5gdy4jrf6zs125wjz0b574g5av";
   };
 
-  postPatch = ''
-    # These paths will be set in the wrapper.
-    sed -e "/export IBUS_DATAROOTDIR/ s/^.*$//" \
-        -e "/export IBUS_LIBEXECDIR/ s/^.*$//" \
-        -e "/export IBUS_LOCALEDIR/ s/^.*$//" \
-        -e "/export IBUS_PREFIX/ s/^.*$//" \
-        -i "setup/ibus-setup.in"
+  patchPhase = ''
+    # https://github.com/NixOS/nixpkgs/issues/30265#issuecomment-335768590
+    substituteInPlace setup/ibus-setup.in --subst-var-by PYTHON ${python3Runtime.interpreter}
+    substituteInPlace data/dconf/Makefile.am --replace "dconf update" true
+    substituteInPlace configure.ac \
+      --replace 'pkgpython2dir="$python2dir/ibus"' \
+                 pkgpython2dir=$out/${python3.sitePackages}/ibus
+    echo \#!/bin/sh > data/dconf/make-dconf-override-db.sh
+    cp ${gtk_doc}/share/gtk-doc/data/gtk-doc.make .
   '';
 
+  preAutoreconf = "touch ChangeLog";
+  preConfigure = "intltoolize";
+
   configureFlags = [
     "--disable-gconf"
-    "--enable-dconf"
     "--disable-memconf"
-    "--enable-ui"
-    "--enable-python-library"
+    (enableFeature (dconf != null) "dconf")
+    (enableFeature (libnotify != null) "libnotify")
+    (enableFeature withWayland "wayland")
+    (enableFeature enablePythonLibrary "python-library")
+    (enableFeature enableUI "ui")
     "--with-unicode-emoji-dir=${emojiData}"
     "--with-emoji-annotation-dir=${cldrEmojiAnnotation}/share/unicode/cldr/common/annotations"
   ];
 
-  buildInputs = [
-    pyEnv
-    intltool isocodes pkgconfig
-    gtk2 gtk3 dconf
-    json_glib
-    dbus libnotify gobjectIntrospection wayland
+  nativeBuildInputs = [
+    autoreconfHook
+    gconf
```

Comment:
Why are you adding `gconf` when it is disabled by `configureFlags`?

Reply:
It's still needed by the build system, unfortunately. It isn't pulled into the closure, though. Disabling `gconf` removes it from the closure, but not from the build.

Reply:
```
<jtojnar> yegortimoshenko: any idea why gconf is suddenly required?  [04:33]
<yegortimoshenko> jtojnar: maybe for some tools?
<jtojnar> seems to be called from here
	  https://github.com/ibus/ibus/blame/master/data/dconf/Makefile.am#L54
								        [04:35]
<yegortimoshenko> right  [04:36]
<yegortimoshenko> $ find /nix/store -name gsettings-schema-convert
<yegortimoshenko>
		 /nix/store/78a9m0l4x60b0l70nl9q5w7v226dwb1a-gconf-3.2.6/bin/gsettings-schema-convert
```

Reply:
Apparently, the GSettings schema generated from the GConf one is listed in `EXTRA_DIST` so it is included in the dist tarball. Since we now no longer use dist tarball, we need to generate it ourselves.

https://github.com/ibus/ibus/pull/1813